### PR TITLE
BB-241 Forbid transitioning/replicating a cold object

### PR DESCRIPTION
**Pull request by nicolas2bert**

### CODE REVIEW

**Comment by nicolas2bert** [comment]
ping

**Review by ghost — APPROVED** [review]
LGTM

File: extensions/lifecycle/tasks/LifecycleTask.js [hunk 1]
```diff
@@ -895,6 +897,13 @@ class LifecycleTask extends BackbeatTask {
             next =>
                 this._getObjectMD(params, log, next),
             (objectMD, next) => {
+                const dataStoreName = objectMD.getDataStoreName();
```
Comment (naren-rajendran) [inline extensions/lifecycle/tasks/LifecycleTask.js:903]: nit: Turn this into an util fn maybe?

**Review by rachedbenmustapha — DISMISSED** [review]

**Comment by rachedbenmustapha** [comment]
@nicolas2bert I missed the case previously when an object is temporarily restored: its `dataStoreName` is not the cold location, so an extra check to take into account the presence of the `archive.restoreCompletedAt` attribute would be good when deciding not to transition the object.

**Review by alexanderchan-scality — APPROVED** [review]

**Pull request merged**